File: HackerRank/Algo/jumpingOnClouds.cc

#include <bits/stdc++.h>

using namespace std;

int jumpingOnClouds(vector<int> c) {
  int jumpCount[102] = {0, 1};
  for (int i = 2; i < (int) c.size(); ++i) {
    if (c[i] == 0) jumpCount[i] = min(jumpCount[i - 2], jumpCount[i - 1]) + 1;
    else jumpCount[i] = 102;
  }
  return jumpCount[(int) c.size() - 1];
}

File: HackerRank/Algo/jumpingOnClouds.java
package hackerrank;

import java.util.List;

class Result {

    /*
     * Complete the 'jumpingOnClouds' function below.
     *
     * The function is expected to return an INTEGER.
     * The function accepts INTEGER_ARRAY c as parameter.
     */

    public static int jumpingOnClouds(List<Integer> c) {
        // Write your code here
        int[] jumpCount = new int[102];
        jumpCount[0] = 0;
        jumpCount[1] = 1;

        for (int i = 2; i < c.size(); ++i) {
            if (c.get(i) == 1) continue;
            if (c.get(i - 2) != 1) jumpCount[i] = jumpCount[i - 2] + 1;

            if (c.get(i - 1) != 1) {
                if (c.get(i - 2) != 1) jumpCount[i] = Math.min(jumpCount[i - 2] + 1, jumpCount[i - 1] + 1);
                else jumpCount[i] = jumpCount[i - 1] + 1;
            }
        }
        return jumpCount[c.size() - 1];
    }
}